oci/validation.go

```diff
@@ -0,0 +1,67 @@
+package oci
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"github.com/opencontainers/image-spec/schema"
+	"github.com/pkg/errors"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+var (
+	fs = schema.FS(false)
+
+	specs = map[string]string{
+		schema.MediaTypeManifest:     "image-manifest-schema.json",
+		schema.MediaTypeManifestList: "manifest-list-schema.json",
+	}
+)
+
+// ValidationError contains all the errors that happened during validation.
+type ValidationError struct {
+	Errs []error
+}
+
+func (e ValidationError) Error() string {
+	return fmt.Sprintf("%v", e.Errs)
+}
+
+// ValidateManifest validates the contents from the given reader as a manifest.
+func ValidateManifest(src io.Reader) error {
```

Comment:
Might be better to declare `Validate` method as part of a var/const instance. This way, validation can be something like this:

```go
if err := MediaTypeManifest.Validate(rd); err != nil { /*...*/ }
```

This still uses the table lookup, but avoids having a method for every single type.